### ISSUE TITLE
Request: Utilize NDesk.Options For Options Parsing

### ISSUE THREAD

**Issue by aolszowka** [aolszowka/MsBuildProjectReferenceDependencyGraph#7]
tl;dr; Use http://www.ndesk.org/Options to perform our options parsing. Its MIT Licensed which is compatible with the license of this project and is extremely lightweight.

Currently we do a bunch of manual options parsing, however this makes it difficult to extend this program by forcing us to deal with command line parsing which isn't our goal. Rather we should spin this out into a library.

There are lots of options parsing libraries available to us, for example Microsoft provides https://github.com/dotnet/command-line-api. However if you look at the number of dependencies that has to carry around you nope out of that pretty quickly.

Rather I am going to suggest we use NDesk.Options for parsing, its a super lightweight dependency (just a single CS file), the usage is very straight forward, and is MIT Licensed.

The only downsides I can see is that it is not actively maintained (last updated back in 2008) and has some pretty gnarly LINQ in it to perform its magic. This probably means that there are some corner cases but at this point its better than rolling our own.
